### PR TITLE
Create Azure Service Principal from CLI

### PR DESCRIPTION
**Pull request by erzetpe**

### CODE REVIEW

**Review by mkyc — DISMISSED** [review]
LGTM .... but I co-authored ... so shouldn't count ;)

File: pkg/az/az_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+	name = "epiphany-cli-tests-" + generateRandomString(6)
+	t.Logf("integration tests will use name %s as created application name", name)
+	return
+}
+
+func TestCreateServicePrincipal(t *testing.T) {
+	//given
+	name, subscriptionID, tenantID := setup(t)
+	pass := generateRandomString(10)
```
Comment (seriva) [inline pkg/az/az_integration_test.go:38]: GeneratePassword(....)?

File: pkg/az/az.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/google/uuid"
+	"github.com/sethvargo/go-password/password"
+)
+
+const (
+	cloudName = "AzurePublicCloud"
+	roleName  = "Contributor"
```
[the remark this answers is not in the document]
Comment (mkyc) [reply pkg/az/az.go:23]: I believe you are right, but that might change flow of creation and I think we should move it to another task. I'll create it now.

File: pkg/az/az_integration_test.go (hunk 1 above)
Comment (mkyc) [reply pkg/az/az_integration_test.go:38]: I don't like using main code methods in tests .... but if @erzetpe is ok with that change, I'm not going to oppose :)

Comment (seriva) [reply pkg/az/az_integration_test.go:38]: @mkyc Also a good point. Lets leave as is for now then.

File: pkg/az/az.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+	return
+}
+
+// GeneratePassword generates Service Principal password
+func GeneratePassword(length, numDigits int) (string, error) {
+	logger.Debug().Msgf("will generate password of length %d with %d digits", length, numDigits)
+	if numDigits > length {
+		return "", fmt.Errorf("parameter 'numDigits' cannot be greater than parameter 'length'")
+	}
+	pass, err := password.Generate(length, numDigits, 0, false, false)
```
[the remark this answers is not in the document]
Comment (erzetpe) [reply pkg/az/az.go]: Changed to also include safe symbols.

File: pkg/az/az_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	if len(tenantID) == 0 {
+		t.Fatalf("expected non-empty TENANT_ID environment variable")
+	}
+
+	subscriptionID = os.Getenv("SUBSCRIPTION_ID")
+	if len(subscriptionID) == 0 {
+		t.Fatalf("expected non-empty SUBSCRIPTION_ID environment variable")
+	}
+
+	name = "epiphany-cli-tests-" + generateRandomString(6)
```
[the remark this answers is not in the document]
Comment (erzetpe) [reply pkg/az/az_integration_test.go:30]: Nice suggestion, but I will add this as TODO as this can be an overkill for this purpose.

**Review by seriva — APPROVED** [review]

**Pull request merged**